tasks/install-matlab/v1/src/matlab.ts

```diff
@@ -21,22 +21,21 @@ export async function makeToolcacheDir(release: Release, platform: string): Prom
     } else {
         if (platform === "win32") {
             toolpath = await windowsHostedToolpath(release).catch(async () => {
-                return await defaultToolpath(release, platform);
+                return await defaultToolpath(release);
             });
         } else {
-            toolpath = await defaultToolpath(release, platform);
+            toolpath = await defaultToolpath(release);
         }
     }
+    if (platform === "darwin") {
+        toolpath = toolpath + "/MATLAB.app";
+    }
     return [toolpath, alreadyExists];
 }
 
-export async function defaultToolpath(release: Release, platform: string): Promise<string> {
+export async function defaultToolpath(release: Release): Promise<string> {
     fs.writeFileSync(".keep", "");

```

Comment:
I should know this, but how does caching this empty keep file end up caching all of MATLAB again? Are we just using this to get the path to a directory within the tool cache and then installing directly into that directory if it doesn't exist? Using essentially the nonexistent function `createCacheEmptyFolderFor("MATLAB", release)`?

Other than that question, everything looks good to me!
:shipit:

Reply:
Yeah exactly, we're creating a toolcache entry to install MATLAB into later. It looks like there's some undocumented stuff surrounding the toolcache so we're just using this as a way to get there using the documented API

Reply:
Gotcha! I might add a comment there explaining the workaround if I remember the next time I work on Azure because it's not super self-evident (to me at least lol) and that might help future maintainers. Or just help me when I forget again in a month 😆